src/Domain/Movie/MovieRepository.php

```diff
@@ -898,6 +898,42 @@ public function fetchWithDirector(int $personId, int $userId) : array
         );
     }
 
+
+    public function fetchFromWatchlistWithActor(int $personId, int $userId) : array
+    {
+        return $this->dbConnection->fetchAllAssociative(
+            <<<SQL
+            SELECT DISTINCT m.*, mur.rating as userRating
+            FROM movie m
+            JOIN movie_cast mc ON m.id = mc.movie_id
+            JOIN person p ON mc.person_id = p.id
+            JOIN watchlist wl ON m.id = wl.movie_id
+            LEFT JOIN movie_user_rating mur ON wl.movie_id = mur.movie_id and mur.user_id = ?
+            WHERE p.id = ? AND m.id IN (SELECT DISTINCT movie_id FROM watchlist) AND wl.user_id = ?
```

Comment:
```suggestion
            WHERE p.id = ? AND m.id IN (wl.movies_id) AND wl.user_id = ?
```

No need for this extra subquery

Reply:
I just tested this but it is giving errors: `[previous exception] [object] (Exception(code: 0): Unable to prepare statement: 1, no such column: wl.movies_id at /app/vendor/doctrine/dbal/src/Driver/SQLite3/Connection.php:25)
`

Reply:
ups, I had a type, it is `movie_id` and not `movie_ids`, it should work now